### PR TITLE
22.2 assign user component

### PR DESCRIPTION
**Pull request by Artem672**

### CODE REVIEW

File: src/app/presentation/components/workspace-components/task-components/workspace-task-assign/workspace-task-assign.component.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    private dataShare: DataShareService,
+    private workspaceService: WorkspaceService,
+    private taskServise: TaskService) {
+   }
+
+  ngOnInit() {
+    this.dataShare.workspaceInfo.subscribe((data : WorkspaceInfo) => {
+      this.workspaceId = data.id;
+    })
+    this.workspaceService.getWorkspaceUserList(this.workspaceId).subscribe((data: WorkspaceMembers[]) =>{
```
Comment (YevheniiVovk) [inline src/app/presentation/components/workspace-components/task-components/workspace-task-assign/workspace-task-assign.component.ts]: More then 100 ch

File: src/app/presentation/components/workspace-components/task-components/workspace-task-assign/workspace-task-assign.component.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+      this.workspaceMembers = data;
+    });
+    this.taskServise.getWorkerRole().subscribe((role: TaskWorkerRole[]) => {
+      this.taskRoles = role;
+    });
+  }
+
+  assignMember()
+  {
```
Comment (YevheniiVovk) [inline src/app/presentation/components/workspace-components/task-components/workspace-task-assign/workspace-task-assign.component.ts]: { should be after method name

File: src/app/presentation/components/workspace-components/task-components/workspace-task-assign/workspace-task-assign.component.ts [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    if(this.assignMembers.length < this.workspaceMembers.length)
+    {
```
Comment (YevheniiVovk) [inline src/app/presentation/components/workspace-components/task-components/workspace-task-assign/workspace-task-assign.component.ts]: same here

File: src/app/presentation/components/workspace-components/task-components/workspace-task-assign/workspace-task-assign.component.ts [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+  }
+
+  deAssignMember(i: number) {
+    this.assignMembers.splice(i, 1);
+  }
+
+  contains(userId: string) : boolean
+  {
+      return this.assignMembers.some(e => e.userId == userId);
+  }
```
Comment (YevheniiVovk) [inline src/app/presentation/components/workspace-components/task-components/workspace-task-assign/workspace-task-assign.component.ts]: { should be after method name

File: src/app/presentation/components/workspace-components/task-components/workspace-task-assign/workspace-task-assign.component.ts [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+  changeUserNameIsValid(isValid: boolean | null){
+    if(isValid)
+    {
```
Comment (YevheniiVovk) [inline src/app/presentation/components/workspace-components/task-components/workspace-task-assign/workspace-task-assign.component.ts]: Same

File: src/app/presentation/components/workspace-components/task-components/workspace-task-assign/workspace-task-assign.component.ts [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+      this.userNameValid = isValid;
+      this.isValid.next(this.userNameValid && this.roleValid);
+    }  
+  }
+
+  changeRoleIsValid(isValid: boolean | null){
```
Comment (YevheniiVovk) [inline src/app/presentation/components/workspace-components/task-components/workspace-task-assign/workspace-task-assign.component.ts]: Whitespace after ) and before {

File: src/app/presentation/components/workspace-components/task-components/workspace-task-assign/workspace-task-assign.component.ts [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+    if(isValid)
+    {
```
Comment (YevheniiVovk) [inline src/app/presentation/components/workspace-components/task-components/workspace-task-assign/workspace-task-assign.component.ts]: { should be after method name

File: src/app/presentation/components/workspace-components/task-components/workspace-task-create/workspace-task-create.component.ts [hunk 8]
```diff
@@ -40,37 +37,20 @@ export class WorkspaceTaskCreateComponent implements OnInit {
...
 
   CreateNewTask(): void{
-    if(this.assignedMembers.some(e => e.userId == null || e.roleTagId == null))
-    {
-      this.alertService.errorMessage("Fill all fields");
-      return;
-    }
-    console.log(this.assignedMembers)
-    if(this.taskForm.valid){
+    if(this.taskForm.valid && this.isAssignedValid){
```
Comment (YevheniiVovk) [inline src/app/presentation/components/workspace-components/task-components/workspace-task-create/workspace-task-create.component.ts]: Whitespace

**Review by herakros — APPROVED** [review]

**Review by FeloniukM — APPROVED** [review]

**Review by FeloniukM — APPROVED** [review]

**Pull request merged**